exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/compliant/CompliantTextRecordReader.java

```diff
@@ -118,12 +118,21 @@ public boolean apply(@Nullable SchemaPath path) {
    * @param outputMutator  Used to create the schema in the output record batch
    * @throws ExecutionSetupException
    */
+  @SuppressWarnings("resource")
   @Override
   public void setup(OperatorContext context, OutputMutator outputMutator) throws ExecutionSetupException {
 
     oContext = context;
-    readBuffer = context.getManagedBuffer(READ_BUFFER);
-    whitespaceBuffer = context.getManagedBuffer(WHITE_SPACE_BUFFER);
+    // Note: DO NOT use managed buffers here. They remain in existence
+    // until the fragment is shut down. The buffers here are large.
```

Comment:
I think the reason you chose to use context.getAllocator() was you don't want to fragmentize managed buffer?
Otherwise you might just call readBuffer.close()? Was there any problem with managed buffer's release? Just curious about the "DO NOT use managed buffer here" part. Besides that, +1.